src/sentry/relay/config/metric_extraction.py

```diff
@@ -210,7 +210,10 @@ def _bulk_cache_query_key(project: Project) -> str:
 
 def _get_bulk_cached_query(project: Project) -> dict[str, Any]:
     query_bulk_cache_key = _bulk_cache_query_key(project)
-    return cache.get(query_bulk_cache_key, None)
+    cache_result = cache.get(query_bulk_cache_key, None)
+
+    with sentry_sdk.start_span(op="on_demand_metrics.query_cache", description=bool(cache_result)):
```

Comment:
should you do set_tag instead?
```suggestion
    with sentry_sdk.set_tag("on_demand_metrics.query_cache", bool(cache_result)):
```

Reply:
Ah that's a good idea Will, I'll make that change in a new commit